File: Leetcode/Array/ConcatenationOfArray.java
package Leetcode.Array;

public class ConcatenationOfArray {

    public int[] getConcatenation(int[] nums) {
        int len = nums.length;
        int[] ans = new int[len * 2];

        for (int i = 0; i < len; i++) {
            ans[i] = nums[i];
        }

        //We are adding second nums array after the first array
        //So we need to i + len to make sure the index is right
        for (int i = 0; i < len; i++) {
            ans[i + len] = nums[i];
        }

        return ans;
    }
}

File: Leetcode_Python/List/concatenation_of_array.py
from typing import List

def getConcatenation(self, nums: List[int]) -> List[int]:
    length = len(nums)
    newNums = [0] * (length * 2)

    for i in range(length):
        newNums[i] = nums[i]
        newNums[i + length] = nums[i]

    return newNums